lightgbm/src/main/scala/com/microsoft/azure/synapse/ml/lightgbm/dataset/DatasetAggregator.scala

```diff
@@ -437,12 +437,12 @@ private[lightgbm] abstract class BaseSparseAggregatedColumns(chunkSize: Int)
   def getIndexPointers: IntSwigArray = indexPointers
 
   override def cleanup(): Unit = {
-    labels.delete()
+    Option(labels).foreach(_.delete())
```

Comment:
in case they are removed, does some error occur?  Just wondering if this is a precautionary thing or it's a bugfix.

Reply:
yes it is, and it's a trick Mark suggested.  labels is not an Option[T], and Mark didn't like the original "== null" comparison

Reply:
hmm discussed offline, let's keep this as is for now to make it simple, but I do think this can be refactored more by not creating the BaseAggregatedColumns in the first place for this case with eval data for the non-main worker

Reply:
hmm discussed offline, let's keep this as is for now to make it simple, but I do think this can be refactored more by not creating the BaseAggregatedColumns in the first place for this case with eval data for the non-main worker